service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderControllerTest.java

```diff
@@ -0,0 +1,213 @@
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.model.C21Order;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.C21OrderBundle;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.service.CreateC21OrderService;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+import uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C21;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
+import static uk.gov.hmcts.reform.fpl.utils.ResourceReader.readBytes;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(C21OrderController.class)
+@OverrideAutoConfiguration(enabled = true)
+class C21OrderControllerTest {
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final String C21_DOCUMENT_TITLE = C21.getDocumentTitle();
+
+    @MockBean
+    private CreateC21OrderService createC21OrderService;
+
+    @MockBean
+    private DocmosisDocumentGeneratorService docmosisDocumentGeneratorService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    void aboutToStartShouldReturnErrorsWhenFamilymanNumberIsNotProvided() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of("data", "some data"))
+                .build())
+            .build();
+
+        MvcResult response = makeRequest(request, "about-to-start");
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = objectMapper.readValue(
+            response.getResponse().getContentAsByteArray(), AboutToStartOrSubmitCallbackResponse.class);
+
+        assertThat(callbackResponse.getErrors()).containsOnlyOnce("Enter Familyman case number");
+    }
+
+    @Test
+    void midEventShouldGenerateC21OrderDocument() throws Exception {
```

Comment:
test names should follow the format SHOULD do this thing WHEN this thing happens

Reply:
inspired by the likes of 'midEventShouldGenerateDraftStandardDirectionDocument()'. but I can try renaming this to follow SHOULD WHEN if you want.  i'd like if we actually followed a convention, its very confusing when i see so many different names for tests code -  i dont see much consistency at all tbh. some tests start with a field name, some with an endpoint, some with should, some with given... and the ones with Should don't always have a When either

Reply:
hmm I agree test names are all over the place, we've got to start somewhere though! I agree that maybe here this is fine, I think other integration test are like this also. Maybe a little harsh of me... but still, always try and follow should and when for good test naming practice :)